### PR TITLE
Add notification for azd failures

### PR DESCRIPTION
**Pull request by danieljurek**
Adds notification configurations for azd pipelines. 

Example successful pipeline run: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1821283&view=logs&j=bc477995-0c4a-574b-536f-2e0d94a7118e&t=330da002-0be6-5a2c-2b14-ef5f270c12c9&l=191

### CODE REVIEW

File: eng/pipelines/notifications.yml [hunk 1]
```diff
@@ -30,6 +30,8 @@ stages:
           PathPrefix: 'cpp'
         Go:
           PathPrefix: 'go'
+        AzDev:
+          PathPrefix: 'azure-accelerators'
```
Comment (weshaggard) [inline eng/pipelines/notifications.yml:34]: Is this how we have named the pipelines? We may need to shorten this to something like azd because I know the team names can get a little long.

**Review by weshaggard — APPROVED** [review]

File: eng/pipelines/notifications.yml (hunk 1 above)
Comment (benbp) [reply eng/pipelines/notifications.yml:34]: Why not just `AzDev`?

Comment (danieljurek) [reply eng/pipelines/notifications.yml:34]: `PathPrefix` is the `azure-accelerators` part of `internal\azure-accelerators\azure-dev - etc.etc.` ... I'm going to change it to `azure-dev` to match the pipeline names. It'll match the first part of the branded name ([Azure Developer CLI](https://docs.microsoft.com/en-us/azure/developer/azure-developer-cli/overview?tabs=nodejs))
